x-pack/plugins/cloud_security_posture/server/create_transforms/create_transforms.ts

```diff
@@ -85,19 +85,27 @@ export const startTransformIfNotStarted = async (
     const transformStats = await esClient.transform.getTransformStats({
       transform_id: transformId,
     });
+
     if (transformStats.count <= 0) {
       logger.error(`Failed starting transform ${transformId}: couldn't find transform`);
       return;
     }
+
     const fetchedTransformStats = transformStats.transforms[0];
-    if (fetchedTransformStats.state === 'stopped') {
+
+    // trying to restart the transform in case it comes to a full stop or failure
+    if (fetchedTransformStats.state === 'stopped' || fetchedTransformStats.state === 'failed') {
       try {
         return await esClient.transform.startTransform({ transform_id: transformId });
       } catch (startErr) {
         const startError = transformError(startErr);
-        logger.error(`Failed starting transform ${transformId}: ${startError.message}`);
+        logger.error(
+          `Failed to start transform ${transformId}. Transform State: Transform State: ${fetchedTransformStats.state}. Error: ${startError.message}`
+        );
       }
-    } else if (
+    }
+
+    if (
       fetchedTransformStats.state === 'stopping' ||
       fetchedTransformStats.state === 'aborting' ||
       fetchedTransformStats.state === 'failed'
```

Comment:
so, this condition `fetchedTransformStats.state === 'failed'` can be removed from here?

Reply:
i removed it initially but then added it back because i wasn't sure, we might want to still get that error message when the state is `failed`. the other message is indicating failing to start which means an attempt to start was made. while this one looks like the general error message. but if you mentioned it as well i guess its better to remove it